module/chunks/chunkVerifier.go

```diff
@@ -69,13 +75,16 @@ func (fcv *ChunkVerifier) SystemChunkVerify(vc *verification.VerifiableChunkData
 	tx := fvm.Transaction(txBody, uint32(0))
 	transactions := []*fvm.TransactionProcedure{tx}
 
-	return fcv.verifyTransactions(vc.Chunk, vc.ChunkDataPack, vc.Result, vc.Header, transactions, vc.EndState)
+	systemChunkContext := fvm.NewContextFromParent(fcv.systemChunkCtx,
```

Comment:
Can we create the `systemChunkCtx` at this point, instead of in `NewChunkVerifier`? since it's not needed if the chunk is not system chunk. 

We could remove the systemChunkCtx field?

Reply:
> Nice work! Could you point our which change is to fix the missing cache hint?

https://github.com/onflow/flow-go/pull/329/files#diff-b4e3be012ea5724c7d1faf9e1ffa2780351d5c876037ccd25ee33704d587e676R296-R304

Pretty much exactly what we already do for program loaded from cache.

Reply:
Not sure if I understand correctly, but we create another context for every new block.

Reply:
Does this work?
```suggestion
	systemChunkContext := makeSystemContext(fcv.vmCtx, vc.Header)
	
	...
	func makeSystemContext(parent fvm.Context, header <Header>) fvm.Context {
	  return fvm.NewContextFromParent(parent,
			fvm.WithRestrictedAccountCreation(false),
			fvm.WithRestrictedDeployment(false),
			fvm.WithTransactionProcessors(fvm.NewTransactionInvocator(parent.Logger),
			fvm.WithBlockHeader(vc.Header))
	}
```

Reply:
It will work, but looks less clear to me. 
Currently we set all "constant" values while initializing `fcv` and only create sub-contexts with new data